docs/source/subscriptions.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Adding Subscriptions To Schema
+---
+
+Adding GraphQL subscriptions to your graphQL schema is simple. since Subscriptions are enabled by default in Apollo 2.0 and you dont' need to install anymore any extra packages.
+As example we want to display to display a list of post that contains author and comment (Query) and we want to add a post to them (Mutation).
+
+Our schema will be:
+
+```
+    type Query {
+        posts: [Post]
+    }
+    type Mutation {
+        addPost(author: String, comment: String): Post
+    }
+    type Post {
+        author: String
+        comment: String
+    }
+```
+
+Our resolver:
+```js
+
+const postResolver = {
+    Query: {
+        posts(root: any, args: any, context: any) {
+            return postController.posts();
+        }
+    },
+    Mutation: {
+        addPost(root: any, args: any, context: any) {
+            pubsub.publish(POST_ADDED, { postAdded: args });
+            return postController.addPost(args);
+        }
+    }
+};
+
+export default postResolver;
+
+```
+
+To enable subscription we need to apply just few changes:
+
+- In our schema:
+
+```
+    type Query {
+        posts: [Post]
+    }
+    type Mutation {
+        addPost(author: String, comment: String): Post
+    }
+    type Post {
+        author: String
+        comment: String
+    }
+    type Subscription {
+        postAdded: Post
+    }
+```
+
+We just added `type Subscription`,
+
+- In the resolver:
+```js
+const POST_ADDED = 'POST_ADDED';
+
+const postResolver = {
+    Subscription: {
+        postAdded: {
+            subscribe: () => pubsub.asyncIterator([POST_ADDED])
+        }
+    },
+    Query: {
+		posts(root: any, args: any, context: any) {
+			return postController.posts();
+		}
+    },
+    Mutation: {
+        addPost(root: any, args: any, context: any) {
+            pubsub.publish(POST_ADDED, { postAdded: args });
+            return postController.addPost(args);
+        }
+    }
+};
+
+export default postResolver;
+```
+1. we added in the mutation `pubsub.publish`
+2. we added the Subscription `postAdded`
+
+In our server we don't need to make any big changes:
+1. we need to export
+```js
+export const pubsub = new PubSub();
+```
+2. In case you defined a context remember to add an extra check related to `req.headers` because context will pass the connection in the form of:
+
+```js
+context: ({ connection }) => { ... }`
+```
+A good workaround to this problem is to return in case `req` or `req.headers` are not defined:
+```js
+const server = new ApolloServer({
+	schema,
+	context: async ({ req }: any) => {
+		if (!req || !req.headers) {
+      		return;
+    	}
+		const token = req.headers.authorization || "";
+		....
+		return checkToken;
+	},
+});
+```
+
+As you can see apollo server 2.0 make the life super easy and allow to have realtime data without change a lot in our existing code.
```

Comment:
Let's make this into "Apollo Server 2.0 allows realtime data without invasive changes to existing code". For some people it may not make life super easy, so we want to be cognizant of those experiences and not assume that it does make life easy

Reply:
applied